dali/pipeline/pipeline.h

```diff
@@ -394,10 +394,13 @@ class DLL_PUBLIC Pipeline {
   DLL_PUBLIC void SaveGraphToDotFile(const std::string &filename, bool show_tensors = false,
                                      bool show_ids = false, bool use_colors = false);
 
+  /// @{
   /**
-   * @brief Returns the batch size that will be produced by the pipeline.
+   * @brief Returns the maximum batch size that can be processed by the Pipeline
    */
-  DLL_PUBLIC inline int batch_size() const { return batch_size_; }
+  DLL_PUBLIC inline int batch_size() const { return max_batch_size_; }
```

Comment:
I can see that this is mostly used in the c_api for obtaining shapes in the SetExternalInput calls. Do you intend to adjust it? It should probably start relying on the inputs to provide the batch size instead of what is stored in the pipeline (and return an error for now).